observability/metric.go

```diff
@@ -1,15 +1,33 @@
 package observability
 
-import "go.uber.org/zap"
+import (
+	"context"
+	"math"
+
+	"go.opentelemetry.io/otel/metric"
+	"go.uber.org/zap"
+)
 
 var (
+	logger                  = zap.L()
 	SecondsHistogramBuckets = []float64{0, 0.001, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10}
 )
 
 func NewMetric[T any](metric T, err error) T {
-	logger := zap.L()
 	if err != nil {
 		logger.Error("failed to instantiate metric", zap.Error(err))
 	}
 	return metric
 }
+
+func RecordUint64Value(ctx context.Context,
+	value uint64,
+	recordF func(ctx context.Context, value int64, options ...metric.RecordOption),
+	options ...metric.RecordOption) {
+	if value > math.MaxInt64 {
+		logger.Warn("value exceeds int64 range, metric not recorded")
+		return
+	}
```

Comment:
Probably better to log it as `Error` not `Warn` since we shouldn't ever see this happen.

Also, I might be tripping but I think in https://github.com/ssvlabs/ssv/pull/1852 I saw `uint32` -> `int32` conversion as well, but you'd probably know better

Reply:
changed log severity to Error. Added unit test